go/libraries/doltcore/schema/typeinfo/linestring.go

```diff
@@ -0,0 +1,180 @@
+	points := make([]types.Point, len(l.Points))
+	for i, p := range l.Points {
+		points[i] = ConvertSQLPointToTypesPoint(p)
+	}
+	return types.Linestring{SRID: l.SRID, Points: points}
+}
+
+// ConvertValueToNomsValue implements TypeInfo interface.
+func (ti *linestringType) ConvertValueToNomsValue(ctx context.Context, vrw types.ValueReadWriter, v interface{}) (types.Value, error) {
+	// Check for null
+	if v == nil {
+		return types.NullValue, nil
+	}
+
+	// Convert to sql.LinestringType
+	line, err := ti.sqlLinestringType.Convert(v)
+	if err != nil {
+		return nil, err
+	}
+
+	return ConvertSQLLinestringToTypesLinestring(line.(sql.Linestring)), nil
+}
+
+// Equals implements TypeInfo interface.
+func (ti *linestringType) Equals(other TypeInfo) bool {
+	if other == nil {
+		return false
+	}
+	if ti2, ok := other.(*linestringType); ok {
+		return ti.sqlLinestringType.Type() == ti2.sqlLinestringType.Type()
+	}
+	return false
+}
+
+// FormatValue implements TypeInfo interface.
+func (ti *linestringType) FormatValue(v types.Value) (*string, error) {
+	if val, ok := v.(types.Linestring); ok {
+		res, err := ti.ConvertNomsValueToValue(val)
+		if err != nil {
+			return nil, err
+		}
+		if resStr, ok := res.(string); ok {
+			return &resStr, nil
+		}
+		return nil, fmt.Errorf(`"%v" has unexpectedly encountered a value of type "%T" from embedded type`, ti.String(), v)
+	}
+	if _, ok := v.(types.Null); ok || v == nil {
+		return nil, nil
+	}
+
+	return nil, fmt.Errorf(`"%v" has unexpectedly encountered a value of type "%T" from embedded type`, ti.String(), v.Kind())
+}
+
+// GetTypeIdentifier implements TypeInfo interface.
+func (ti *linestringType) GetTypeIdentifier() Identifier {
+	return LinestringTypeIdentifier
+}
+
+// GetTypeParams implements TypeInfo interface.
+func (ti *linestringType) GetTypeParams() map[string]string {
+	return map[string]string{}
+}
+
+// IsValid implements TypeInfo interface.
+func (ti *linestringType) IsValid(v types.Value) bool {
+	if val, ok := v.(types.Linestring); ok {
+		_, err := ti.sqlLinestringType.Convert(val)
+		if err != nil {
+			return false
+		}
+		return true
+	}
+	if _, ok := v.(types.Null); ok || v == nil {
+		return true
+	}
+	return false
+}
+
+// NomsKind implements TypeInfo interface.
+func (ti *linestringType) NomsKind() types.NomsKind {
+	return types.LinestringKind
+}
+
+// ParseValue implements TypeInfo interface.
+func (ti *linestringType) ParseValue(ctx context.Context, vrw types.ValueReadWriter, str *string) (types.Value, error) {
+	if str == nil || *str == "" {
+		return types.NullValue, nil
+	}
+	return ti.ConvertValueToNomsValue(context.Background(), nil, *str)
+}
+
+// Promote implements TypeInfo interface.
+func (ti *linestringType) Promote() TypeInfo {
+	return &linestringType{ti.sqlLinestringType.Promote().(sql.LinestringType)}
+}
+
+// String implements TypeInfo interface.
+func (ti *linestringType) String() string {
+	return "Linestring()"
```

Comment:
These don't have any parentheses in the name if there are no parameters to put in there.